### ISSUE TITLE
[BLOG] Getting started as a TAG ENV Contributor

### ISSUE THREAD

**Issue by guidemetothemoon** [cncf/tag-env-sustainability#212]
- **Authors**: *TBD*
- **Reviewers**: *TBD* 
- **TAG ENV Lead**: *TBD* 

## Abstract

This blog post provides a detailed introduction to TAG ENV, what its mission and goals are and why one would consider joining the TAG and contributing to it. In addition it should provide a good overview of how one can contribute, what initiatives (like working groups and events like Cloud Native Sustainability Week) are currently ongoing so that it's easier to navigate for a newcomer that has no previous knowledge about the TAG. 

## Scope

Introductory blog post providing good overview of contribution possibilities to the different areas of the TAG ENV. A good example here is a blog post created by TAG Contributor Strategy last year: https://www.cncf.io/blog/2022/03/10/the-maintainers-toolkit-must-know-resources-for-cncf-projects

## Expected audience

- Newcomers who have no previous knowledge about the TAG ENV
- Recent joiners who struggle understanding where and how they can contribute to TAG ENV

## Goals and Non Goals

Goals:
- Provide introduction to the TAG ENV, its mission and goals
- Provide reasons for why one would consider contributing to the TAG ENV
- Provide overview of areas where one can contribute and how
- Encourage to join the TAG ENV and spread the word about it to reach broader audience

Non-Goals:
- Go into details of different projects or deliverables of the TAG ENV
- Get into technical or scientifical details around cloud native sustainability (this is an introductory blog post that should be inclusive and understandable for both technical and non-technical audience).

## Tasks

A more complete summary of tasks are described on XYZ.

- [ ] Outline Blog post
    - [x] [**Author**] described the abstract of the blog post, defined the scope, goals and non-goals, and defined the expected audience.
    - [x] [**Author**] notified TAG ENV community to review the outline via Slack on the `tag-env-sustainability` channel.
    - [x] [**TAG ENV Lead**] One of the TAG ENV leads assigned themselves to wrangle the blog
    - [x] [**TAG ENV Lead**] Blog post outline and theme has been discussed
- [x] [**Author**] Blog post written and ready for review
- [ ] Review
    - [x] [**Author**] blog post PR opened (blog posts are added to the [`website/content/blog`](https://github.com/cncf/tag-env-sustainability/tree/main/website/content/blog) folder, images to [`website/static/images/blog/<YEAR>-<MONTH>-<title>/*.jpg`](https://github.com/cncf/tag-env-sustainability/tree/main/website/static/images))
    - [ ] [**Author**] notified TAG ENV community for review via Slack on the `tag-env-sustainability` channel.
    - [ ] [**Author**] gave a minimum of a week to collect public feedback, depending on reviews this will get extended. If there is a major event (conference, etc.) or holidays coming up, please extend accordingly.
    - [ ] [**Author**] Feedback and comments have been resolved.
    - [ ] [**Reviewers**] reviewed blog post.
    - [ ] [**TAG ENV Lead**] reviewed blog post.
    - [ ] [**TAG ENV Lead**] ready to get published.
    - [ ] [**TAG ENV Lead**] notified one of the ToC liaisons to sign off the blog post.
    - [ ] [**TAG ENV ToC Liaison**] reviewed blog post.
    - [ ] [**TAG ENV Lead**] merged blog post.
    - [ ] _optional_ [**TAG ENV Lead**] schedule the blog post on the CNCF blog next to the TAG ENV blog
    - [ ] [**Author**] Wrote feedback about the blog post process as a comment under this issue

## Comments 

Document: https://docs.google.com/document/d/1-6FS0JB7qvemXZDKa7bhLEa6-xpJj6HgyTTY_bBN04U/edit?usp=sharing
Somewhat built upon the discussion from this issue: #61

**Comment by ATREAY** [comment]
Hey @guidemetothemoon, I'm new to CNCF tag-env-sustainability. Can I work on this blog?
Also, I have a quick question: once assigned, do I need to complete the entire blog or can I contribute to a specific section?

**Comment by immavalls** [comment]
Hu, I can help with this as well. I am new to the TAG and writing this should help me too.

**Comment by AntonioDiTuri** [comment]
I am also interested in this one!

**Comment by Mfonn** [comment]
Hi @guidemetothemoon I am interested in this issue. 
I recently started contributing to TAG and I write about my experience contributing to Open Source Projects. 
I'd love to work on this issue.
This is an example of my writing: 
https://mfon.hashnode.dev/cncf-zero-to-merge-becoming-a-contributorpart-iii#clmrranmj09ebidnv4dyb5u18

**Comment by ATREAY** [comment]
Hey @guidemetothemoon, thanks for assigning this issue and for answering my question. 
I'll let you know once I've made some updates to the blog.

**Comment by AntonioDiTuri** [comment]
Yes sure! I will be happy to coordinate! What are the next steps?

**Comment by ATREAY** [comment]
Hey @AntonioDiTuri, I believe it's a smart move to split up the tasks for the blog. 
I'm cool with taking on the "Why join TAG ENV?" section. I'll handle that part.

**Comment by guidemetothemoon** [comment]
@AntonioDiTuri fantastic, thank you for taking the lead on this! I've outlined the approximate structure suggestion on the blog post so I think that it makes a lot of sense to spread it across those of you who are interested in contributing ☺️

@ATREAY @immavalls @Mfonn @AntonioDiTuri Some links that can come in handy when it comes to TAG ENV, its mission and goals, as well as contributor guidance:

- TAG ENV mission and goals: https://github.com/cncf/tag-env-sustainability/blob/main/charter.md
- TAG ENV contributor guidance: https://github.com/cncf/tag-env-sustainability/blob/main/charter.md
- TAG ENV contributor path: https://github.com/cncf/tag-env-sustainability/blob/main/governance/contributor-ladder.md
- Description of TAG ENV Working Groups, their mission and goals - currently only Green Reviews WG is there, but Comms WG will be published soon in the same place as well: https://github.com/cncf/tag-env-sustainability/tree/main/working-groups

If you have any questions or require any clarification, please reach out! You can also tag me directly in the doc with guidemetothemoon@gmail.com if I should take a look at any part of the content☺️

Thanks again for your time and effort on this blog post!

**Comment by AntonioDiTuri** [comment]
Thanks for the terrific work! I think that really helps! Looking forward to do this guys! If you find some other really good sources let's share it here! They can help all the community grow awareness. If you write something please let us know! We would love to give feedback!

**Comment by ATREAY** [comment]
Hey folks, I've added some info in the "Why join TAG ENV?" section. Could you take a peek and let me know if it needs any changes?

**Comment by Mfonn** [comment]
@AntonioDiTuri  Thank you for splitting up the work. 
I am comfortable with taking the "What if I have an Idea" section. 
However I would like more context on it. 
Does "Idea" mean adding a new feature to TAG ENV or creating a group or event or something along those lines?
Or does it mean something else?

**Comment by AntonioDiTuri** [comment]
Hi @Mfonn. Have from the document those are the guidelines: 

> TODO: share how folks can come with ideas, suggestions, requests for change. F.ex. what GH issues can be relevant, join the meeting to share it in the group or add it on Slack, link to formal process description, etc.

The purpose of this section should be to explain how everyone can bring in the TAG their proposals and how to discuss them so that they become real. I have an idea: how can I interact with the TAG?

Some examples: 
- proposal: https://github.com/cncf/tag-env-sustainability/issues/64
- event: https://github.com/cncf/tag-env-sustainability/issues/106, and more in general everything related to the cloud native sustainability week https://github.com/orgs/cncf/projects/10/views/11 
- groups: in this context we talk mainly about working groups, take a look at Green Reviews WG: https://github.com/orgs/cncf/projects/10/views/12

**Comment by Mfonn** [comment]
@AntonioDiTuri Thank you very much for the explanation. 
I wrote a draft that I'll like to be reviewed. 
Here's the link: https://docs.google.com/document/d/1QYnm0Oo084xNhUQfggQ3oxrKHVSIocBK9BrVmlRH2sg/edit?usp=sharing

All feedback would be greatly appreciated, thank you.

**Comment by caradelia** [comment]
@catblade @leonardpahlke @mkorbi could you review the draft above?

**Comment by guidemetothemoon** [comment]
@Mfonn can you please move the draft that you've created to the common doc we're using for this blog post for better overview and so that it's easier for others who are involved to collaborate and provide input? https://docs.google.com/document/d/1-6FS0JB7qvemXZDKa7bhLEa6-xpJj6HgyTTY_bBN04U/edit?usp=sharing

Thanks in advance!

**Comment by guidemetothemoon** [comment]
@ATREAY thank you for adding content to the "Why join" section. I think we will need to wait a little bit for the rest of the content to be added so that it can be reviewed as a whole with regards to style, story flow, language, etc. I will therefore await a little bit with reviewing your section until we get the first version of the full draft ready.

@AntonioDiTuri I was wondering if we should put a timeframe for this blog post so that it's measurable and we can be able to progress. It could be really great to get this blog post published prior to KubeCon + CloudNativeCon North America in November - could this be realistic?

**Comment by ATREAY** [comment]
Sure thing @guidemetothemoon! It's all good, and having a timeline makes sense. It'll help ensure we finish the blog on schedule, and the reviewers will have enough time to go over it thoroughly without rushing.

**Comment by AntonioDiTuri** [comment]
Hi @guidemetothemoon sorry for the delay, yes sure! 

Then I would say let's set: 
- **writing** deadline for **Friday 27th October**
- **reviewing** deadline for **Friday 3th November**
So that we can go live on the 6th! What do you guys say?


Status: 
@ATREAY: "Why join TAG ENV?" -> written, anyone available for review?
@immavalls the "How can I get involved?" -> to write
@Mfonn the "What if I have an idea?" -> written, anyone available for review?
@AntonioDiTuri intro and conclusion -> to write

@immavalls I am waiting for your part to not repeat things too much in intro and conclusion, could you notify me when you are done with writing?

**Comment by guidemetothemoon** [comment]
@immavalls since we haven't heard from you regarding your capacity and availability to contribute to this blog post I'm wondering if you're OK if we assign someone else to take the blog post section that we thought that you could write? Please let us know before Wednesday, 1st of November if you still want to write it. Please note that our goal is to publish the blog post before the end of KubeCon+CloudNativeCon NA so we can't postpone the completion too much. Thanks in advance for your feedback!

@AntonioDiTuri if @immavalls will not be able to write the "How can I get involved?" section can you, @ATREAY or @Mfonn take it this week? If not, I can hop in and write it as well Thursday or Friday this week, but I would like to give you this opportunity first.

If we get all the sections written this week I'm ready to do the overall review as soon as it's ready.

**Comment by AntonioDiTuri** [comment]
Thanks @guidemetothemoon for giving me this opportunity. 
I can submit my parts and the "How can I get involved section" by Thursday.  
Is it ok?

**Comment by guidemetothemoon** [comment]
Sounds good, thanks @AntonioDiTuri! Once it's ready I can start the review process.

**Comment by AntonioDiTuri** [comment]
Ready!

**Comment by catblade** [comment]
Where is the PR?  Would be nice if we could get this in before KubeCon, but it is short notice.

**Comment by guidemetothemoon** [comment]
@AntonioDiTuri @ATREAY @Mfonn great work you all! Thank you so much for putting your time and effort into writing this blog post and coordinating your collective efforts in this. It's shaping really nicely now and I strongly believe that we can get it published in the beginning of next week. I extended the intro-section a bit more and added some minor suggestions and comments here and there - please check it out and let me know if you require any further clarifications.

@catblade @leonardpahlke additional pair of review eyes would be highly appreciated!
@patricia-cahill @caradelia  FYI, blog post PR is coming soon.

Link to the doc for quick access: https://docs.google.com/document/d/1-6FS0JB7qvemXZDKa7bhLEa6-xpJj6HgyTTY_bBN04U/edit

**Comment by leonardpahlke** [comment]
Added some comments

**Comment by AntonioDiTuri** [comment]
Thanks for the comment and the review I have accepted most of them related to my parts. 
@ATREAY @Mfonn please take a look at your parts respectively so that we can respect the deadline :) 
Let me know if anything else is needed

**Comment by AntonioDiTuri** [comment]
@Mfonn there are several comments pending on your section. Could you please take a look?

**Comment by catblade** [comment]
@AntonioDiTuri  it looks really good.  Added in some comments around wording updates to make more engaging and to remove extra words, but it's very good.  Thank you.

**Comment by guidemetothemoon** [comment]
@Mfonn @ATREAY  @AntonioDiTuri  Thank you all for making updates!💚

@AntonioDiTuri can you take a look at the comments and if those look OK to accept then I would suggest to merge/resolve those and just submit a PR with the blog so that we take any remaining small modifications there. Can you please create a PR if you have time?🙌 

Here's an example PR of an earlier blog post submission that you can use for reference: https://github.com/cncf/tag-env-sustainability/pull/242

Please note that every PR has markdown linting validation so you may need to fix eventual findings from it so that the build is green, but I can help you with that once the PR is created.

**Comment by guidemetothemoon** [comment]
Blog is finalized and published: https://tag-env-sustainability.cncf.io/blog/2023-getting-started-as-tag-env-contributor
Closing this issue - great work from everyone involved 💚
